### ISSUE TITLE
replacing azurerm_traffic_manager_endpoint by azurerm_traffic_manager_azure_endpoint throws Error: no schema available for azurerm_traffic_manager_endpoint.ep while reading state

### ISSUE THREAD

**Issue by alexeyauslender** [hashicorp/terraform-provider-azurerm#16539]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.9

### AzureRM Provider Version

3.3.0

### Affected Resource(s)/Data Source(s)

azurerm_traffic_manager_endpoint

### Terraform Configuration Files

```hcl
before upgrading azurerm 
resource "azurerm_traffic_manager_endpoint" "ep" {
  for_each = local.apsvc_map_with_locations

  name                = "${azurerm_app_service.webapp[each.key].name}-ep"
  resource_group_name = var.non_regional_web_rg.name

  profile_name       = azurerm_traffic_manager_profile.tm[each.value.apsvc_name].name
  type               = "azureEndpoints"
  target_resource_id = azurerm_app_service.webapp[each.key].id
  priority           = each.value.ep_priority
}

after upgrading azurerm to 3.3.0
resource "azurerm_traffic_manager_azure_endpoint" "ep_azure" {
  for_each           = local.apsvc_map_with_locations
  name               = "${azurerm_app_service.webapp[each.key].name}-ep"
  profile_id         = azurerm_traffic_manager_profile.tm[each.value.apsvc_name].id
  priority           = each.value.ep_priority
  target_resource_id = azurerm_app_service.webapp[each.key].id
  weight             = 1000
}
```

### Debug Output/Panic Output

```shell
2022-04-25T18:19:56.3613199Z Error: no schema available for module.web.azurerm_traffic_manager_endpoint.ep["eastus2-showcase"] while reading state; this is a bug in Terraform and should be reported
2022-04-25T18:19:56.3613688Z 
2022-04-25T18:19:56.3614614Z 
2022-04-25T18:19:56.3624575Z 
2022-04-25T18:19:56.3625431Z Error: no schema available for module.web.azurerm_traffic_manager_endpoint.ep["centralus-showcase"] while reading state; this is a bug in Terraform and should be reported
2022-04-25T18:19:56.3625860Z 
2022-04-25T18:19:56.3626120Z 
2022-04-25T18:19:56.7309746Z ##[error][ d:\_wf\04\r269\a\.terraform\terraform.exe plan -var sub=app508 -var pod_name=aa -var build_number=1.0.22097.1 -var env=Development -var dayforce_universe=Development ] 
2022-04-25T18:19:56.7315788Z ##[error]Error: no schema available for module.web.azurerm_traffic_manager_endpoint.ep["eastus2-showcase"] while reading state; this is a bug in Terraform and should be reported
2022-04-25T18:19:56.7317631Z ##[error]Error: no schema available for module.web.azurerm_traffic_manager_endpoint.ep["centralus-showcase"] while reading state; this is a bug in Terraform and should be reported
```

### Expected Behaviour

The old resource deleted and the new one is created instead

### Actual Behaviour

Error: no schema available for module.web.azurerm_traffic_manager_endpoint.ep["eastus2-showcase"] while reading state; this is a bug in Terraform and should be reported

### Steps to Reproduce

upgrade azurerm provider to 3.3.0
replace azurerm_traffic_manager_endpoint
by azurerm_traffic_manager_azure_endpoint

**Comment by magodo** [comment]
@alexeyauslender Thank you for submitting this!

The error is due to you directly renamed the resource type while the state still holds the old resource type, while the old resource type's schema no longer exist in the new provider version.

In this case, you can run `terraform state rm azurerm_traffic_manager_endpoint.ep` to remove these resources from state, then do the change you've done above in the config. Next, you'll need to import these resources one by one, and tune the config a bit to make sure `terraform plan` doesn't return a plan diff.

**Comment by alexeyauslender** [comment]
Hi @magodo, I was trying to run terraform state rm and here is the error that I've got 
Failed to marshal state to json: no schema found for azurerm_traffic_manager_endpoint.ep
Basically exactly the same error as I am getting during the regular run when azurerm_traffic_manager_endpoint was replaced by azurerm_traffic_manager_azure_endpoint

**Comment by magodo** [comment]
@alexeyauslender Exactly, it is mentioned here: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_traffic_manager_endpoint.

So it seems the last resort is to downgrade the provider version back to v2.99.0, do the `terraform state rm`, then upgrade the provider and do the importing...

**Comment by tanerjn** [comment]
With the version trials(2, 2.99, 3.77) this did not really work. What is the best actual practice to use Traffic Manager?
